lib/vrl/stdlib/src/mod_func.rs

```diff
@@ -0,0 +1,107 @@
+use ::value::Value;
+use vrl::prelude::*;
+
+fn r#mod(value: Value, modulus: Value) -> Resolved {
+    let result = value.try_rem(modulus)?;
+    Ok(result)
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Mod;
+
+impl Function for Mod {
+    fn identifier(&self) -> &'static str {
+        "mod"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::INTEGER | kind::FLOAT,
+                required: true,
+            },
+            Parameter {
+                keyword: "modulus",
+                kind: kind::INTEGER | kind::FLOAT,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "mod",
+            source: r#"mod(5, 3)"#,
+            result: Ok("2"),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        mut arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let modulus = arguments.required("modulus");
+        // TODO: return a compile-time error if modulus is 0
```

Comment:
Looks like this has been left in unintentionally 😄

Reply:
This was intentional, from the discussion here: https://github.com/vectordotdev/vector/pull/14081#discussion_r954533838